src/Propel/Generator/Reverse/MysqlSchemaParser.php

```diff
@@ -431,7 +431,9 @@ protected function addPrimaryKey(Table $table)
                 continue;
             }
             $name = $row['Column_name'];
-            $table->getColumn($name)->setPrimaryKey(true);
+            if ($column = $table->getColumn($name)) {
```

Comment:
Good catch!

Could you please avoid var assignment and condition on the same line?

MariaDB is not the only MySQL engine Propel supports.
Is there a test, that could confirm that other engines are not affected and for MariaDB it's solved?